### PR TITLE
Removing utcnow method deprecated at python3.12

### PR DESCRIPTION
**Pull request by rmad17**
Replacing `datetime.utcnow()` due to deprecation as discussed in #2407

### CODE REVIEW

**Comment by sentrivana** [comment]
Thanks for the PR @rmad17! This works for Python 3.12 as well as older Python 3 versions, but we need to support 2.7 too, where `datetime.timezone` doesn't exist -- there we still need to keep doing it the old way.

My suggestion would be to add a new `utcnow()` function to the [_compat.py file](https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/_compat.py) that does `datetime.utcnow()` for py2.7 and `datetime.now(timezone.utc)` for py3+, and use that function everywhere.

**Review by sentrivana — CHANGES_REQUESTED** [review]
See comment above 🙂

**Comment by rmad17** [comment]
Thanks for the suggestion @sentrivana. I have implemented what you suggested.

File: sentry_sdk/_compat.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import sys
 import contextlib
+from datetime import datetime, timezone
```
Comment (sentrivana) [inline sentry_sdk/_compat.py:3]: If we try to import `timezone` here it'll raise an ImportError in Python 2 -- I'd only import `datetime` here...
```suggestion
from datetime import datetime
```
...and then `timezone` in the `else` branch of the `if PY2:` condition that starts on line 85, e.g. before the `urllib.parse` import.

**Comment by hynek** [comment]
Should this also fix the same warnings around `datetime.datetime.utcfromtimestamp` or do y'all prefer a separate PR?

It seems to be called only once AFAICT:

https://github.com/getsentry/sentry-python/blob/59a67d329420857168ff26dc78a2a9a003be89e9/sentry_sdk/integrations/logging.py#L285

**Review by sentrivana — CHANGES_REQUESTED** [review]
Looks good, just one more thing to make it work in py2!

**Comment by sentrivana** [comment]
> Should this also fix the same warnings around `datetime.datetime.utcfromtimestamp` or do y'all prefer a separate PR?
> 
> It seems to be called only once AFAICT:
> 
> https://github.com/getsentry/sentry-python/blob/59a67d329420857168ff26dc78a2a9a003be89e9/sentry_sdk/integrations/logging.py#L285

Ah, good catch. Don't really care much if it gets its own PR.

@rmad17 if you also feel like fixing what @hynek pointed out, feel free to add to this PR (not necessary though!). Otherwise I'll open a new issue for it.

**Comment by rmad17** [comment]
Thanks @sentrivana for the suggestions. I have made the changes. I have also addressed the issue @hynek pointed out.

**Review by sentrivana — APPROVED** [review]
Thank you for squeezing in the `utcfromtimestamp` change as well @rmad17!

I added some typing-related stuff to make the linters happy. We should be good to go!

**Comment by rmad17** [comment]
Thanks @sentrivana  for merging the changes:)

**Pull request merged**